google/cloud/vision_helpers/__init__.py

```diff
@@ -26,7 +27,7 @@ class VisionHelpers(object):
     See the :class:`~google.cloud.vision_v1.ImageAnnotatorClient`.
     """
 
-    def annotate_image(self, request, retry=None, timeout=None):
+    def annotate_image(self, request, retry=None, timeout=None, metadata=()):
```

Comment:
The new logic for this method doesn't feel quite right. I'll do another, more in depth pass later today, but I think that the type checking logic hides some potentially confusing new behavior.

Reply:
The original code supports `dict` and the classes defined in [`image_annotator_pb2.py`](https://github.com/googleapis/python-vision/blob/master/google/cloud/vision_v1/proto/image_annotator_pb2.py), but not `proto.Message`. Also the two cases `# If the image is a file handler, set the content.` and `# If a filename is provided, read the file.` don't look like relevant to `proto.Message`. I did some tweaks to make it also work for `proto.Message`.